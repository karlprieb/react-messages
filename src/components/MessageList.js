import React from 'react'
import PropTypes from 'prop-types'
import Message from './Message'

const MessageList = ({ messages }) => {
  const messageCount = messages.length
  const allMessages = messages.map(message =>
    <Message
      key={message.id}
      {...message}
    />
  )

  return (
    <ul>
      {allMessages}
    </ul>
  )
}

MessageList.propTypes = {
  messages: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    text: PropTypes.string.isRequired,
    owner: PropTypes.string.isRequired
  }).isRequired).isRequired
}

export default MessageList
