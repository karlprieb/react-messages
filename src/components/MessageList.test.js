import React from 'react'
import MessageList from './MessageList'

const messagesMock = [
  {
    id: 0,
    text: 'Test message 0',
    owner: 'yours'
  },
  {
    id: 1,
    text: 'Another test message with id 1',
    owner: 'theirs'
  },
  {
    id: 2,
    text: 'Message with id 2',
    owner: 'yours'
  }
]

describe('MessageList component', () => {
  it('should render a list of messages', () => {
    const wrapper = shallow(<MessageList messages={messagesMock} />)
    expect(wrapper).toMatchSnapshot()
  })
})
