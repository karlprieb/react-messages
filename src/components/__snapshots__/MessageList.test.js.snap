// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList component should render a list of messages 1`] = `
<ul>
  <Message
    id={0}
    key="0"
    owner="yours"
    text="Test message 0"
  />
  <Message
    id={1}
    key="1"
    owner="theirs"
    text="Another test message with id 1"
  />
  <Message
    id={2}
    key="2"
    owner="yours"
    text="Message with id 2"
  />
</ul>
`;
